### PR TITLE
IntelliSense fixes - PR 12/?

### PR DESCRIPTION
**Pull request by mairaw**
- Adds missing descriptions
- Adds dropped exception conditions during migration
- Style guide fixes
- Minor fixes

Contributes to #1330

### CODE REVIEW

File: xml/System.Collections.Generic/SynchronizedReadOnlyCollection`1.xml [hunk 1]
```diff
@@ -1001,9 +975,7 @@
       </ReturnValue>
       <Docs>
         <summary>This property returns <see langword="true" /> because the read-only collection is always fixed in size.</summary>
-        <value>Returns <see cref="T:System.Boolean" />.
```
Comment (rpetrusha) [inline xml/System.Collections.Generic/SynchronizedReadOnlyCollection`1.xml]: I wonder if there are a lot more of these remaining in the documentation...

File: xml/System.ServiceModel.Channels/MsmqBindingElementBase.xml [hunk 2]
```diff
@@ -457,7 +457,11 @@
   
  ]]></format>
         </remarks>
-        <exception cref="T:System.ArgumentOutOfRangeException">The value is not infinite and larger than Int32.MaxValue milliseconds.</exception>
+        <exception cref="T:System.ArgumentOutOfRangeException">The value is less than zero.
+        
+-or-
+
+The value is not infinite and larger than Int32.MaxValue milliseconds.</exception>
```
Comment (rpetrusha) [inline xml/System.ServiceModel.Channels/MsmqBindingElementBase.xml]: nit for a future fix: "The value is not infinite and is larger than Int32.MaxValue milliseconds."
There are a few occurrences of this.

File: xml/System.ServiceModel.Channels/MsmqTransportBindingElement.xml [hunk 3]
```diff
@@ -100,7 +100,48 @@
...
+
+UseActiveDirectory is true and QueueTransferProtocol is not Native.
+
+-or-
+
+CustomDeadLetterQueue is not null and DeadLetterQueue is not Custom.
+
+-or-
+
+DeadLetterQueue is set to Custom on Windows 2003/XP.
```
Comment (rpetrusha) [inline xml/System.ServiceModel.Channels/MsmqTransportBindingElement.xml]: Also for a future fix: I think this one can be deleted.

File: xml/System.ServiceModel.Channels/MsmqTransportBindingElement.xml [hunk 4]
```diff
@@ -100,7 +100,48 @@
...
+
+-or-
+
+ExactlyOnce is false and DeadLetterQueue is Custom.
+
+-or-
+
+CustomDeadLetterQueue is set and if there is no write permission to write to queue or if queue does not exist.
```
Comment (rpetrusha) [inline xml/System.ServiceModel.Channels/MsmqTransportBindingElement.xml]: also for a future fix: ...and either there is no write permission to write to the queue, or the queue does not exist."

File: xml/System.ServiceModel.Channels/MsmqTransportBindingElement.xml [hunk 5]
```diff
@@ -100,7 +100,48 @@
...
+
+-or-
+
+CustomDeadLetterQueue is non-transactional.
+
+-or-
+
+MsmqTransportSecurity.MsmqAuthenticationMode is set to WindowsDomain and if MSMQ is installed without ActiveDirectory integration.
```
Comment (rpetrusha) [inline xml/System.ServiceModel.Channels/MsmqTransportBindingElement.xml]: also for a future fix: "if" should be deleted.

File: xml/System.ServiceModel.Dispatcher/XPathMessageFilter.xml [hunk 6]
```diff
@@ -215,7 +215,12 @@
           <paramref name="reader" /> is <see langword="null" />.</exception>
         <exception cref="T:System.ArgumentException">
           <paramref name="reader" /> is not positioned on the starting element.</exception>
-        <exception cref="T:System.InvalidOperationException">An invalid dialect attribute was read: it was not XPath.</exception>
+        <exception cref="T:System.InvalidOperationException">
+          There were problems reading the dialect attribute, namespace URI, or duplicate attributes were found.
```
Comment (rpetrusha) [inline xml/System.ServiceModel.Dispatcher/XPathMessageFilter.xml]: also for a future fix: "dialect attribute or namespace URI, or...

File: xml/System.ServiceModel.MsmqIntegration/MsmqIntegrationBindingElement.xml [hunk 7]
```diff
@@ -89,7 +89,39 @@
...
-          <see cref="P:System.ServiceModel.MsmqTransportSecurity.MsmqAuthenticationMode" /> is Certificate or WindowsDomain and <see cref="P:System.ServiceModel.MsmqTransportSecurity.MsmqProtectionLevel" /> is None.</exception>
+          <paramref name="ExactlyOnce" /> is <see langword="true" /> and <paramref name="Durable" /> is <see langword="false" />.
+
+-or-
+
+CustomDeadLetterQueue is not null and DeadLetterQueue is not Custom.
+
+-or-
+
+DeadLetterQueue is set to Custom on Windows 2003/XP.
```
Comment (rpetrusha) [inline xml/System.ServiceModel.MsmqIntegration/MsmqIntegrationBindingElement.xml]: for a future fix: can be deleted. (?)

File: xml/System.ServiceModel.MsmqIntegration/MsmqIntegrationBindingElement.xml [hunk 8]
```diff
@@ -135,7 +167,43 @@
         <exception cref="T:System.ArgumentNullException">The context value is <see langword="null" />.</exception>
         <exception cref="T:System.ArgumentException">The <paramref name="TChannel" /> is not <see langword="IInputChannel" />.</exception>
         <exception cref="T:System.InvalidOperationException">
-          <see cref="P:System.ServiceModel.MsmqTransportSecurity.MsmqAuthenticationMode" /> is Certificate or WindowsDomain and <see cref="P:System.ServiceModel.MsmqTransportSecurity.MsmqProtectionLevel" /> is None.</exception>
+          ExactlyOnce is <see langword="true" /> and Durable is <see langword="false" />.
+
+-or-
+
+Queue not readable.
```
Comment (rpetrusha) [inline xml/System.ServiceModel.MsmqIntegration/MsmqIntegrationBindingElement.xml]: for a future fix: The queue is not readable.

**Comment by mairaw** [comment]
Thanks for the comments @rpetrusha. I’ll address them tomorrow.

Yeah, I just copied the messages as-is. I’d need to research whether the condition still applies to newer OSes.

Re. the return sections, I think we still have tons. I was going to open an issue to take care of that in the future. I’ve been fixing the ones that are popping in my tests.

**Pull request merged**